Dockerfile

```diff
@@ -1,50 +1,61 @@
 # syntax=docker/dockerfile:1
 
-ARG BASE_IMG_TAG=nonroot
+ARG GO_VERSION="1.18"
+ARG RUNNER_IMAGE="gcr.io/distroless/static:nonroot"
 
 # --------------------------------------------------------
-# Build 
+# Builder
 # --------------------------------------------------------
 
-FROM golang:1.18.2-alpine3.15 as build
+FROM golang:${GO_VERSION}-alpine as builder
 
-# linux-headers needed by github.com/zondax/hid
 RUN set -eux; apk add --no-cache ca-certificates build-base; apk add git linux-headers
 
-# CosmWasm: see https://github.com/CosmWasm/wasmvm/releases
-# 1) Add the releases into /lib/
-ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.0.0/libwasmvm_muslc.aarch64.a \ 
-  https://github.com/CosmWasm/wasmvm/releases/download/v1.0.0/libwasmvm_muslc.x86_64.a \ 
-  /lib/
-# 2) Verify their hashes
-RUN sha256sum /lib/libwasmvm_muslc.aarch64.a | grep 7d2239e9f25e96d0d4daba982ce92367aacf0cbd95d2facb8442268f2b1cc1fc; \
-  sha256sum /lib/libwasmvm_muslc.x86_64.a | grep f6282df732a13dec836cda1f399dd874b1e3163504dbd9607c6af915b2740479
-# Copy the right library according to architecture. The final location will be found by the linker flag `-lwasmvm_muslc`
-RUN cp /lib/libwasmvm_muslc.$(uname -m).a /lib/libwasmvm_muslc.a
-
-# Get go dependencies first
-COPY go.mod go.sum /osmosis/
+# Download go dependencies
 WORKDIR /osmosis
+COPY go.mod go.mod
+COPY go.sum go.sum
```

Comment:
Consider doing this in one command to minimize the number of layers. As per:
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers

Reply:
Thanks! I'add in the next PR I'm making so we don't have to review again the changes